packages/serde/src/artifact.ts

```diff
@@ -84,37 +88,52 @@ export function decodeArtifactPointValues(
   };
 }
 
-export type RawArtifactWithMetadata = Awaited<ReturnType<DarkForest['getArtifactById']>>;
+function calculateByteUInt(tokenId: EthersBN, startByte: number, endByte: number) {
+  const token = utils.arrayify(tokenId);
+  let byteUInt = 0;
+  for (let i = startByte; i <= endByte; i++) {
+    byteUInt += token[i] * 256 ** (endByte - i);
+  }
+  return byteUInt;
+}
 
 /**
- * Converts the raw typechain result of `ArtifactTypes.ArtifactWithMetadata`
- * struct to an `Artifact` typescript typed object (see @dfdao/types).
+ * Converts the raw Token ID to an `Artifact` typescript typed object (see @dfdao/types).
  *
- * @param rawArtifactWithMetadata Raw data of an `ArtifactWithMetadata` struct,
- * returned from a blockchain call (assumed to be typed with typechain).
+ * @param tokenId Raw `tokenId` representing an `Artifact` struct
  */
-export function decodeArtifact(rawArtifactWithMetadata: RawArtifactWithMetadata): Artifact {
-  const { artifact, owner, upgrade, timeDelayedUpgrade, locationId, voyageId } =
-    rawArtifactWithMetadata;
+export function decodeArtifact(tokenId: EthersBN): Artifact {
+  // 454093553060631688131761916536684786657829610989267258091508080987871379456
+  // Collection: Artifact
+  // Rarity: Common
+  // Type: Colossus
+  // Biome: Forest
+
+  // These acocunt for unknown at the 0-th index
+  const tokenIdx = ArtifactInfo.TokenType - 1;
+  const rarityIdx = ArtifactInfo.ArtifactRarity - 1;
+  const typeIdx = ArtifactInfo.ArtifactType - 1;
+  const biomeIdx = ArtifactInfo.Biome - 1;
+
+  const _tokenType = calculateByteUInt(tokenId, tokenIdx, tokenIdx);
```

Comment:
What should happen if something tries to decode an invalid token type? I think we should probably throw in the client too. WDYT?